dockstore-common/src/main/java/io/dockstore/common/PipHelper.java

```diff
@@ -14,7 +14,8 @@
  */
 public final class PipHelper {
     public static final String DEV_SEM_VER = "development-build";
-    public static final String SEM_VER_STRING = "(^(\\d+)\\.(\\d+)\\.(\\d+)$)|(^" + DEV_SEM_VER + "$)";
+    // Regex copied from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+    public static final String SEM_VER_STRING = "(^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$)|(^" + DEV_SEM_VER + "$)";
```

Comment:
You should fix the Sonar cloud bug otherwise we will just need to fix it later on. Here is an explanation of why catastrophic backtracking happens https://www.regular-expressions.info/catastrophic.html. A way to fix is here https://www.regular-expressions.info/possessive.html. You can test your fix here https://regex101.com/. It's likely you will just need to replace * with '*+'